sympy/solvers/ode/single.py

```diff
@@ -791,66 +791,93 @@ def _get_general_solution(self, *, simplify: bool = True):
         return [gensol]
 
 
-class SecondNonlinearAutonomousConserved(SinglePatternODESolver):
+class SecondAutonomousNth(SinglePatternODESolver):
     r"""
-    Gives solution for the autonomous second order nonlinear
-    differential equation of the form
+    The general second order autonomous differential equation of the form:
+    .. math :: x''(t) + f(x)*x'(t) + g(x) = 0
 
-    .. math :: f''(x) = g(f(x))
+    This solver is added for the special case in this genre:
+    .. math :: f''(x) = g(f(x)) f'(x)^n
 
-    The solution for this differential equation can be computed
-    by multiplying by `f'(x)` and integrating on both sides,
-    converting it into a first order differential equation.
+    This special case can be solved using by writing the second derivative in
+    form involving a power of x' and on rewriting the second derivative, rearranging.
+
+    For proof of the results, kindly refer the below link:
+    https://en.wikipedia.org/wiki/Autonomous_system_(mathematics)
+    http://eqworld.ipmnet.ru/en/solutions/ode/ode0301.pdf
+
+    Caution: If n is even except for 2, kindly set simplify = False in dsolve()
```

Comment:
This needs an explanation

Reply:
When `n` is even except for 2, it has a pair of solutions, so if we don't set `simplify` as `False`, it goes in the `_remove_redundant_solutions` which can't handle three equations in one soln i.e. in `soln1` in case of `f(x).diff(x, x) + f(x).diff(x)**4`, moreover,  `_remove_redundant_solutions` only handles simple cases.
Kindly give your opinion.
Thanks.

Reply:
So if `n` is a symbol that is not known to be even or odd then I guess we can always return the pair and let `_remove_redundant_solutions` try to handle it. I think it's fine to sometimes return redundant solutions if necessary. Returning a Piecewise should be avoided unless it really is unavoidable.